{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3eefb940",
   "metadata": {
    "papermill": {
     "duration": 0.011521,
     "end_time": "2023-07-04T21:36:08.013950",
     "exception": false,
     "start_time": "2023-07-04T21:36:08.002429",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c5dad48",
   "metadata": {
    "papermill": {
     "duration": 0.008724,
     "end_time": "2023-07-04T21:36:08.032574",
     "exception": false,
     "start_time": "2023-07-04T21:36:08.023850",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c1774ace",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2023-07-04T21:36:08.045107Z",
     "iopub.status.busy": "2023-07-04T21:36:08.044811Z",
     "iopub.status.idle": "2023-07-04T21:36:12.416842Z",
     "shell.execute_reply": "2023-07-04T21:36:12.416362Z"
    },
    "papermill": {
     "duration": 4.378952,
     "end_time": "2023-07-04T21:36:12.418074",
     "exception": false,
     "start_time": "2023-07-04T21:36:08.039122",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9af778bd",
   "metadata": {
    "papermill": {
     "duration": 0.002347,
     "end_time": "2023-07-04T21:36:12.422178",
     "exception": false,
     "start_time": "2023-07-04T21:36:12.419831",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2d117970",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2023-07-04T21:36:12.425908Z",
     "iopub.status.busy": "2023-07-04T21:36:12.425683Z",
     "iopub.status.idle": "2023-07-04T21:36:12.428695Z",
     "shell.execute_reply": "2023-07-04T21:36:12.428392Z"
    },
    "papermill": {
     "duration": 0.005994,
     "end_time": "2023-07-04T21:36:12.429635",
     "exception": false,
     "start_time": "2023-07-04T21:36:12.423641",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'ana-flavia-superlactacao.yaml')\n",
    "experiment_name = 'jenneffer-vs-01'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'redemicro')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'not-hist-vs-manifest.csv')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'imgs'))\n",
    "replace_files = False\n",
    "trim = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c176e263",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-07-04T21:36:12.433174Z",
     "iopub.status.busy": "2023-07-04T21:36:12.433054Z",
     "iopub.status.idle": "2023-07-04T21:36:12.436012Z",
     "shell.execute_reply": "2023-07-04T21:36:12.435599Z"
    },
    "papermill": {
     "duration": 0.005741,
     "end_time": "2023-07-04T21:36:12.436865",
     "exception": false,
     "start_time": "2023-07-04T21:36:12.431124",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"miliane-CxAC-trim\"\n",
    "base_dir = \"/mnt/nupeb/rede-micro/redemicro-miliane-nutri\"\n",
    "manifest_file = \"/mnt/nupeb/rede-micro/redemicro-miliane-nutri/data/raw/manifest/miliane-manifest-CxAC.csv\"\n",
    "metadata_file = \"/mnt/nupeb/rede-micro/redemicro-miliane-nutri/data/raw/metadata/miliane-metadata-CxAC.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = (\n",
    "    \"/mnt/nupeb/rede-micro/qiime2-classifiers/silva-138-99-nb-classifier.qza\"\n",
    ")\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1eec1865",
   "metadata": {
    "papermill": {
     "duration": 0.001481,
     "end_time": "2023-07-04T21:36:12.439840",
     "exception": false,
     "start_time": "2023-07-04T21:36:12.438359",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ac1687fe",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2023-07-04T21:36:12.443394Z",
     "iopub.status.busy": "2023-07-04T21:36:12.443274Z",
     "iopub.status.idle": "2023-07-04T21:36:12.446547Z",
     "shell.execute_reply": "2023-07-04T21:36:12.446250Z"
    },
    "papermill": {
     "duration": 0.00608,
     "end_time": "2023-07-04T21:36:12.447444",
     "exception": false,
     "start_time": "2023-07-04T21:36:12.441364",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cabc2e5e",
   "metadata": {
    "papermill": {
     "duration": 0.00152,
     "end_time": "2023-07-04T21:36:12.450497",
     "exception": false,
     "start_time": "2023-07-04T21:36:12.448977",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d6e1461e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-07-04T21:36:12.454128Z",
     "iopub.status.busy": "2023-07-04T21:36:12.454008Z",
     "iopub.status.idle": "2023-07-04T21:36:12.458815Z",
     "shell.execute_reply": "2023-07-04T21:36:12.458412Z"
    },
    "papermill": {
     "duration": 0.007665,
     "end_time": "2023-07-04T21:36:12.459702",
     "exception": false,
     "start_time": "2023-07-04T21:36:12.452037",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3832d046",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2023-07-04T21:36:12.463353Z",
     "iopub.status.busy": "2023-07-04T21:36:12.463233Z",
     "iopub.status.idle": "2023-07-04T21:44:00.224790Z",
     "shell.execute_reply": "2023-07-04T21:44:00.224281Z"
    },
    "papermill": {
     "duration": 467.764781,
     "end_time": "2023-07-04T21:44:00.226030",
     "exception": false,
     "start_time": "2023-07-04T21:36:12.461249",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        d_type, \n",
    "        manifest_file, \n",
    "        view_type=v_type)\n",
    "    \n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "16cb5c56",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.602855Z",
     "start_time": "2022-06-28T19:44:58.642520Z"
    },
    "execution": {
     "iopub.execute_input": "2023-07-04T21:44:00.230237Z",
     "iopub.status.busy": "2023-07-04T21:44:00.230101Z",
     "iopub.status.idle": "2023-07-04T21:44:00.241793Z",
     "shell.execute_reply": "2023-07-04T21:44:00.241373Z"
    },
    "papermill": {
     "duration": 0.014746,
     "end_time": "2023-07-04T21:44:00.242670",
     "exception": false,
     "start_time": "2023-07-04T21:44:00.227924",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de937295",
   "metadata": {
    "papermill": {
     "duration": 0.001589,
     "end_time": "2023-07-04T21:44:00.245824",
     "exception": false,
     "start_time": "2023-07-04T21:44:00.244235",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e2162974",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2023-07-04T21:44:00.249641Z",
     "iopub.status.busy": "2023-07-04T21:44:00.249450Z",
     "iopub.status.idle": "2023-07-04T21:44:00.252122Z",
     "shell.execute_reply": "2023-07-04T21:44:00.251637Z"
    },
    "papermill": {
     "duration": 0.005604,
     "end_time": "2023-07-04T21:44:00.253018",
     "exception": false,
     "start_time": "2023-07-04T21:44:00.247414",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: f8382eb5-41f8-4188-a086-5ae23bb70390>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "90d11c71",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2023-07-04T21:44:00.256847Z",
     "iopub.status.busy": "2023-07-04T21:44:00.256722Z",
     "iopub.status.idle": "2023-07-04T21:44:00.261617Z",
     "shell.execute_reply": "2023-07-04T21:44:00.261307Z"
    },
    "papermill": {
     "duration": 0.007762,
     "end_time": "2023-07-04T21:44:00.262461",
     "exception": false,
     "start_time": "2023-07-04T21:44:00.254699",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/lauro/data/f8382eb5-41f8-4188-a086-5ae23bb70390')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: f8382eb5-41f8-4188-a086-5ae23bb70390>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1fa58942",
   "metadata": {
    "papermill": {
     "duration": 0.00177,
     "end_time": "2023-07-04T21:44:00.265986",
     "exception": false,
     "start_time": "2023-07-04T21:44:00.264216",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4d669a83",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-07-04T21:44:00.270115Z",
     "iopub.status.busy": "2023-07-04T21:44:00.269991Z",
     "iopub.status.idle": "2023-07-04T21:45:36.382193Z",
     "shell.execute_reply": "2023-07-04T21:45:36.381717Z"
    },
    "papermill": {
     "duration": 96.115568,
     "end_time": "2023-07-04T21:45:36.383293",
     "exception": false,
     "start_time": "2023-07-04T21:44:00.267725",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121711_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121711_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121711_0_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121711_1_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121711_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121711_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121711_0_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121711_1_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.280 s (5.625 µs/read; 10.67 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             49,793\n",
      "  Read 1 with adapter:                  48,368 (97.1%)\n",
      "  Read 2 with adapter:                  45,264 (90.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            5,807 (11.7%)\n",
      "Pairs written (passing filters):        43,986 (88.3%)\n",
      "\n",
      "Total basepairs processed:    25,394,430 bp\n",
      "  Read 1:    15,186,865 bp\n",
      "  Read 2:    10,207,565 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     20,506,528 bp (80.8%)\n",
      "  Read 1:    12,516,130 bp\n",
      "  Read 2:     7,990,398 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 48368 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.8\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t18\t0.0\t0\t18\n",
      "18\t7506\t0.0\t0\t7506\n",
      "19\t8677\t0.0\t0\t8677\n",
      "20\t8970\t0.0\t0\t8970\n",
      "21\t7956\t0.0\t0\t7956\n",
      "22\t8261\t0.0\t0\t8261\n",
      "23\t6957\t0.0\t0\t6957\n",
      "24\t10\t0.0\t0\t10\n",
      "25\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 45264 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t2\t0.0\t0\t2\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t15\t0.0\t0\t15\n",
      "21\t8594\t0.0\t0\t8594\n",
      "22\t8129\t0.0\t0\t8129\n",
      "23\t7417\t0.0\t0\t7417\n",
      "24\t7747\t0.0\t0\t7747\n",
      "25\t6894\t0.0\t0\t6894\n",
      "26\t6458\t0.0\t0\t6458\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121712_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121712_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121712_2_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121712_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121712_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121712_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121712_2_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121712_3_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.815 s (4.134 µs/read; 14.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            197,224\n",
      "  Read 1 with adapter:                 191,943 (97.3%)\n",
      "  Read 2 with adapter:                 181,981 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           19,971 (10.1%)\n",
      "Pairs written (passing filters):       177,253 (89.9%)\n",
      "\n",
      "Total basepairs processed:   100,584,240 bp\n",
      "  Read 1:    60,153,320 bp\n",
      "  Read 2:    40,430,920 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     82,634,027 bp (82.2%)\n",
      "  Read 1:    50,437,083 bp\n",
      "  Read 2:    32,196,944 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 191943 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t3\t0.2\t0\t3\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t44\t0.0\t0\t44\n",
      "18\t29908\t0.0\t0\t29908\n",
      "19\t34447\t0.0\t0\t34447\n",
      "20\t35353\t0.0\t0\t35353\n",
      "21\t31785\t0.0\t0\t31785\n",
      "22\t32514\t0.0\t0\t32514\n",
      "23\t27840\t0.0\t0\t27840\n",
      "24\t22\t0.0\t0\t22\n",
      "32\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 181981 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t60\t0.0\t0\t60\n",
      "21\t33812\t0.0\t0\t33812\n",
      "22\t32958\t0.0\t0\t32958\n",
      "23\t30086\t0.0\t0\t30086\n",
      "24\t30787\t0.0\t0\t30787\n",
      "25\t27789\t0.0\t0\t27789\n",
      "26\t26433\t0.0\t0\t26433\n",
      "27\t31\t0.0\t0\t31\n",
      "29\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121713_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121713_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121713_4_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121713_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121713_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121713_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121713_4_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121713_5_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.721 s (4.219 µs/read; 14.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            170,758\n",
      "  Read 1 with adapter:                 165,742 (97.1%)\n",
      "  Read 2 with adapter:                 156,948 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           18,349 (10.7%)\n",
      "Pairs written (passing filters):       152,409 (89.3%)\n",
      "\n",
      "Total basepairs processed:    87,086,580 bp\n",
      "  Read 1:    52,081,190 bp\n",
      "  Read 2:    35,005,390 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     71,052,054 bp (81.6%)\n",
      "  Read 1:    43,367,422 bp\n",
      "  Read 2:    27,684,632 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 165742 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.6\t0\t2\n",
      "9\t3\t0.7\t0\t3\n",
      "10\t3\t0.2\t0\t3\n",
      "11\t7\t0.0\t0\t7\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t44\t0.0\t0\t44\n",
      "18\t25727\t0.0\t0\t25727\n",
      "19\t29769\t0.0\t0\t29769\n",
      "20\t30353\t0.0\t0\t30353\n",
      "21\t27382\t0.0\t0\t27382\n",
      "22\t28371\t0.0\t0\t28371\n",
      "23\t24038\t0.0\t0\t24038\n",
      "24\t18\t0.0\t0\t18\n",
      "25\t2\t0.0\t0\t2\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 156948 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t54\t0.0\t0\t54\n",
      "21\t29379\t0.0\t0\t29379\n",
      "22\t28333\t0.0\t0\t28333\n",
      "23\t25988\t0.0\t0\t25988\n",
      "24\t26572\t0.0\t0\t26572\n",
      "25\t23804\t0.0\t0\t23804\n",
      "26\t22782\t0.0\t0\t22782\n",
      "27\t21\t0.0\t0\t21\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121714_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121714_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121714_6_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121714_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121714_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121714_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121714_6_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121714_7_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.175 s (3.793 µs/read; 15.82 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            309,657\n",
      "  Read 1 with adapter:                 301,273 (97.3%)\n",
      "  Read 2 with adapter:                 284,385 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,799 (10.6%)\n",
      "Pairs written (passing filters):       276,858 (89.4%)\n",
      "\n",
      "Total basepairs processed:   157,925,070 bp\n",
      "  Read 1:    94,445,385 bp\n",
      "  Read 2:    63,479,685 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):    129,071,295 bp (81.7%)\n",
      "  Read 1:    78,780,305 bp\n",
      "  Read 2:    50,290,990 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 301273 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t4.7\t0\t2\n",
      "9\t6\t1.2\t0\t6\n",
      "10\t1\t0.3\t0\t1\n",
      "11\t4\t0.1\t0\t4\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t14\t0.0\t0\t14\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t103\t0.0\t0\t103\n",
      "18\t46760\t0.0\t0\t46760\n",
      "19\t54306\t0.0\t0\t54306\n",
      "20\t55444\t0.0\t0\t55444\n",
      "21\t50194\t0.0\t0\t50194\n",
      "22\t50872\t0.0\t0\t50872\n",
      "23\t43489\t0.0\t0\t43489\n",
      "24\t48\t0.0\t0\t48\n",
      "25\t1\t0.0\t0\t1\n",
      "27\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 284385 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t104\t0.0\t0\t104\n",
      "21\t53159\t0.0\t0\t53159\n",
      "22\t51530\t0.0\t0\t51530\n",
      "23\t47018\t0.0\t0\t47018\n",
      "24\t48091\t0.0\t0\t48091\n",
      "25\t43184\t0.0\t0\t43184\n",
      "26\t41236\t0.0\t0\t41236\n",
      "27\t27\t0.0\t0\t27\n",
      "29\t2\t0.0\t0\t2\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121715_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121715_9_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121715_8_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121715_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121715_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121715_9_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121715_8_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121715_9_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.565 s (3.758 µs/read; 15.97 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            416,343\n",
      "  Read 1 with adapter:                 405,322 (97.4%)\n",
      "  Read 2 with adapter:                 382,754 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           43,507 (10.4%)\n",
      "Pairs written (passing filters):       372,836 (89.6%)\n",
      "\n",
      "Total basepairs processed:   212,334,930 bp\n",
      "  Read 1:   126,984,615 bp\n",
      "  Read 2:    85,350,315 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):    173,816,084 bp (81.9%)\n",
      "  Read 1:   106,092,368 bp\n",
      "  Read 2:    67,723,716 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 405322 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t6.4\t0\t2\n",
      "9\t4\t1.6\t0\t4\n",
      "10\t5\t0.4\t0\t5\n",
      "11\t10\t0.1\t0\t10\n",
      "13\t7\t0.0\t0\t7\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t22\t0.0\t0\t22\n",
      "17\t131\t0.0\t0\t131\n",
      "18\t63635\t0.0\t0\t63635\n",
      "19\t73144\t0.0\t0\t73144\n",
      "20\t74590\t0.0\t0\t74590\n",
      "21\t66828\t0.0\t0\t66828\n",
      "22\t68473\t0.0\t0\t68473\n",
      "23\t58397\t0.0\t0\t58397\n",
      "24\t48\t0.0\t0\t48\n",
      "25\t1\t0.0\t0\t1\n",
      "26\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 382754 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t6\t0.0\t0\t6\n",
      "19\t18\t0.0\t0\t18\n",
      "20\t121\t0.0\t0\t121\n",
      "21\t71493\t0.0\t0\t71493\n",
      "22\t68868\t0.0\t0\t68868\n",
      "23\t63265\t0.0\t0\t63265\n",
      "24\t65274\t0.0\t0\t65274\n",
      "25\t58104\t0.0\t0\t58104\n",
      "26\t55515\t0.0\t0\t55515\n",
      "27\t58\t0.0\t0\t58\n",
      "28\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "32\t2\t0.0\t0\t2\n",
      "45\t1\t0.0\t0\t1\n",
      "51\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n",
      "63\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121716_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121716_11_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121716_10_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121716_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121716_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121716_11_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121716_10_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121716_11_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.918 s (4.158 µs/read; 14.43 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            220,699\n",
      "  Read 1 with adapter:                 214,683 (97.3%)\n",
      "  Read 2 with adapter:                 202,229 (91.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           23,902 (10.8%)\n",
      "Pairs written (passing filters):       196,797 (89.2%)\n",
      "\n",
      "Total basepairs processed:   112,556,490 bp\n",
      "  Read 1:    67,313,195 bp\n",
      "  Read 2:    45,243,295 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     91,746,847 bp (81.5%)\n",
      "  Read 1:    55,999,931 bp\n",
      "  Read 2:    35,746,916 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 214683 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.8\t0\t1\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t4\t0.1\t0\t4\n",
      "12\t4\t0.0\t0\t4\n",
      "14\t11\t0.0\t0\t11\n",
      "15\t11\t0.0\t0\t11\n",
      "16\t12\t0.0\t0\t12\n",
      "17\t57\t0.0\t0\t57\n",
      "18\t33644\t0.0\t0\t33644\n",
      "19\t38542\t0.0\t0\t38542\n",
      "20\t39428\t0.0\t0\t39428\n",
      "21\t35699\t0.0\t0\t35699\n",
      "22\t36513\t0.0\t0\t36513\n",
      "23\t30727\t0.0\t0\t30727\n",
      "24\t26\t0.0\t0\t26\n",
      "25\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 202229 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t60\t0.0\t0\t60\n",
      "21\t37643\t0.0\t0\t37643\n",
      "22\t36409\t0.0\t0\t36409\n",
      "23\t33568\t0.0\t0\t33568\n",
      "24\t34369\t0.0\t0\t34369\n",
      "25\t30608\t0.0\t0\t30608\n",
      "26\t29509\t0.0\t0\t29509\n",
      "27\t34\t0.0\t0\t34\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121717_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121717_13_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121717_12_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121717_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121717_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121717_13_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121717_12_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121717_13_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.575 s (5.113 µs/read; 11.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            112,481\n",
      "  Read 1 with adapter:                 109,357 (97.2%)\n",
      "  Read 2 with adapter:                 102,843 (91.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,454 (11.1%)\n",
      "Pairs written (passing filters):       100,027 (88.9%)\n",
      "\n",
      "Total basepairs processed:    57,365,310 bp\n",
      "  Read 1:    34,306,705 bp\n",
      "  Read 2:    23,058,605 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     46,631,078 bp (81.3%)\n",
      "  Read 1:    28,461,891 bp\n",
      "  Read 2:    18,169,187 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 109357 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t0\t1\n",
      "9\t1\t0.4\t0\t1\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t24\t0.0\t0\t24\n",
      "18\t17241\t0.0\t0\t17241\n",
      "19\t19396\t0.0\t0\t19396\n",
      "20\t19921\t0.0\t0\t19921\n",
      "21\t17857\t0.0\t0\t17857\n",
      "22\t18707\t0.0\t0\t18707\n",
      "23\t16186\t0.0\t0\t16186\n",
      "24\t7\t0.0\t0\t7\n",
      "26\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 102843 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t45\t0.0\t0\t45\n",
      "21\t19092\t0.0\t0\t19092\n",
      "22\t18461\t0.0\t0\t18461\n",
      "23\t17158\t0.0\t0\t17158\n",
      "24\t17425\t0.0\t0\t17425\n",
      "25\t15688\t0.0\t0\t15688\n",
      "26\t14948\t0.0\t0\t14948\n",
      "27\t12\t0.0\t0\t12\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121718_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121718_15_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121718_14_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121718_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121718_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121718_15_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121718_14_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121718_15_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.486 s (4.318 µs/read; 13.90 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            112,618\n",
      "  Read 1 with adapter:                 109,503 (97.2%)\n",
      "  Read 2 with adapter:                 102,994 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,401 (11.0%)\n",
      "Pairs written (passing filters):       100,217 (89.0%)\n",
      "\n",
      "Total basepairs processed:    57,435,180 bp\n",
      "  Read 1:    34,348,490 bp\n",
      "  Read 2:    23,086,690 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     46,718,895 bp (81.3%)\n",
      "  Read 1:    28,515,186 bp\n",
      "  Read 2:    18,203,709 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 109503 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t35\t0.0\t0\t35\n",
      "18\t16903\t0.0\t0\t16903\n",
      "19\t19486\t0.0\t0\t19486\n",
      "20\t20143\t0.0\t0\t20143\n",
      "21\t18144\t0.0\t0\t18144\n",
      "22\t18690\t0.0\t0\t18690\n",
      "23\t16079\t0.0\t0\t16079\n",
      "24\t10\t0.0\t0\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 102994 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t0\t2\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t32\t0.0\t0\t32\n",
      "21\t19334\t0.0\t0\t19334\n",
      "22\t18458\t0.0\t0\t18458\n",
      "23\t17010\t0.0\t0\t17010\n",
      "24\t17502\t0.0\t0\t17502\n",
      "25\t15537\t0.0\t0\t15537\n",
      "26\t15092\t0.0\t0\t15092\n",
      "27\t10\t0.0\t0\t10\n",
      "38\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121719_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121719_17_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121719_16_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121719_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121719_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121719_17_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121719_16_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121719_17_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.588 s (4.275 µs/read; 14.04 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            137,450\n",
      "  Read 1 with adapter:                 133,616 (97.2%)\n",
      "  Read 2 with adapter:                 125,984 (91.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           14,914 (10.9%)\n",
      "Pairs written (passing filters):       122,536 (89.1%)\n",
      "\n",
      "Total basepairs processed:    70,099,500 bp\n",
      "  Read 1:    41,922,250 bp\n",
      "  Read 2:    28,177,250 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     57,126,283 bp (81.5%)\n",
      "  Read 1:    34,867,786 bp\n",
      "  Read 2:    22,258,497 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 133616 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.1\t0\t2\n",
      "9\t1\t0.5\t0\t1\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t34\t0.0\t0\t34\n",
      "18\t21042\t0.0\t0\t21042\n",
      "19\t23916\t0.0\t0\t23916\n",
      "20\t24408\t0.0\t0\t24408\n",
      "21\t21855\t0.0\t0\t21855\n",
      "22\t22715\t0.0\t0\t22715\n",
      "23\t19602\t0.0\t0\t19602\n",
      "24\t17\t0.0\t0\t17\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 125984 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.5\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t42\t0.0\t0\t42\n",
      "21\t23541\t0.0\t0\t23541\n",
      "22\t22951\t0.0\t0\t22951\n",
      "23\t20821\t0.0\t0\t20821\n",
      "24\t21171\t0.0\t0\t21171\n",
      "25\t19072\t0.0\t0\t19072\n",
      "26\t18364\t0.0\t0\t18364\n",
      "27\t6\t0.0\t0\t6\n",
      "29\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121720_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121720_19_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121720_18_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121720_19_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121720_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121720_19_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121720_18_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121720_19_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.449 s (3.897 µs/read; 15.40 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            371,795\n",
      "  Read 1 with adapter:                 361,865 (97.3%)\n",
      "  Read 2 with adapter:                 340,718 (91.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           40,027 (10.8%)\n",
      "Pairs written (passing filters):       331,768 (89.2%)\n",
      "\n",
      "Total basepairs processed:   189,615,450 bp\n",
      "  Read 1:   113,397,475 bp\n",
      "  Read 2:    76,217,975 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):    154,669,634 bp (81.6%)\n",
      "  Read 1:    94,405,636 bp\n",
      "  Read 2:    60,263,998 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 361865 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.7\t0\t1\n",
      "9\t5\t1.4\t0\t5\n",
      "10\t5\t0.4\t0\t5\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t7\t0.0\t0\t7\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t16\t0.0\t0\t16\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t17\t0.0\t0\t17\n",
      "17\t104\t0.0\t0\t104\n",
      "18\t56376\t0.0\t0\t56376\n",
      "19\t65299\t0.0\t0\t65299\n",
      "20\t66080\t0.0\t0\t66080\n",
      "21\t60115\t0.0\t0\t60115\n",
      "22\t61294\t0.0\t0\t61294\n",
      "23\t52470\t0.0\t0\t52470\n",
      "24\t51\t0.0\t0\t51\n",
      "30\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "40\t2\t0.0\t0\t2\n",
      "49\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 340718 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t7\t0.0\t0\t7\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t103\t0.0\t0\t103\n",
      "21\t63764\t0.0\t0\t63764\n",
      "22\t61383\t0.0\t0\t61383\n",
      "23\t56302\t0.0\t0\t56302\n",
      "24\t57686\t0.0\t0\t57686\n",
      "25\t51653\t0.0\t0\t51653\n",
      "26\t49749\t0.0\t0\t49749\n",
      "27\t41\t0.0\t0\t41\n",
      "39\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "61\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121721_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121721_21_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121721_20_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121721_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121721_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121721_21_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121721_20_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121721_21_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.377 s (3.985 µs/read; 15.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            345,448\n",
      "  Read 1 with adapter:                 336,214 (97.3%)\n",
      "  Read 2 with adapter:                 317,650 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           36,167 (10.5%)\n",
      "Pairs written (passing filters):       309,281 (89.5%)\n",
      "\n",
      "Total basepairs processed:   176,178,480 bp\n",
      "  Read 1:   105,361,640 bp\n",
      "  Read 2:    70,816,840 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):    144,189,236 bp (81.8%)\n",
      "  Read 1:    88,006,094 bp\n",
      "  Read 2:    56,183,142 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 336214 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.3\t0\t1\n",
      "9\t3\t1.3\t0\t3\n",
      "10\t2\t0.3\t0\t2\n",
      "11\t6\t0.1\t0\t6\n",
      "12\t8\t0.0\t0\t8\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t14\t0.0\t0\t14\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t16\t0.0\t0\t16\n",
      "17\t84\t0.0\t0\t84\n",
      "18\t52454\t0.0\t0\t52454\n",
      "19\t60710\t0.0\t0\t60710\n",
      "20\t61341\t0.0\t0\t61341\n",
      "21\t56018\t0.0\t0\t56018\n",
      "22\t57176\t0.0\t0\t57176\n",
      "23\t48327\t0.0\t0\t48327\n",
      "24\t29\t0.0\t0\t29\n",
      "26\t1\t0.0\t0\t1\n",
      "31\t2\t0.0\t0\t2\n",
      "33\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "51\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 317650 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.3\t0\t1\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t12\t0.0\t0\t12\n",
      "20\t113\t0.0\t0\t113\n",
      "21\t60040\t0.0\t0\t60040\n",
      "22\t57552\t0.0\t0\t57552\n",
      "23\t52276\t0.0\t0\t52276\n",
      "24\t53977\t0.0\t0\t53977\n",
      "25\t47798\t0.0\t0\t47798\n",
      "26\t45819\t0.0\t0\t45819\n",
      "27\t39\t0.0\t0\t39\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121722_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121722_23_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121722_22_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121722_23_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.16\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121722_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-w6ibsml7/210421121722_23_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121722_22_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/fa2ef6ea-dacc-44dd-8ed2-358980c9d77e/data/210421121722_23_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.504 s (4.207 µs/read; 14.26 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            119,832\n",
      "  Read 1 with adapter:                 116,476 (97.2%)\n",
      "  Read 2 with adapter:                 110,758 (92.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,111 (10.1%)\n",
      "Pairs written (passing filters):       107,721 (89.9%)\n",
      "\n",
      "Total basepairs processed:    61,114,320 bp\n",
      "  Read 1:    36,548,760 bp\n",
      "  Read 2:    24,565,560 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     50,218,813 bp (82.2%)\n",
      "  Read 1:    30,651,862 bp\n",
      "  Read 2:    19,566,951 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 116476 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.5\t0\t2\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t26\t0.0\t0\t26\n",
      "18\t18120\t0.0\t0\t18120\n",
      "19\t21100\t0.0\t0\t21100\n",
      "20\t21249\t0.0\t0\t21249\n",
      "21\t19183\t0.0\t0\t19183\n",
      "22\t19892\t0.0\t0\t19892\n",
      "23\t16869\t0.0\t0\t16869\n",
      "24\t12\t0.0\t0\t12\n",
      "42\t1\t0.0\t0\t1\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 110758 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t4\t0.0\t0\t4\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t42\t0.0\t0\t42\n",
      "21\t20941\t0.0\t0\t20941\n",
      "22\t19733\t0.0\t0\t19733\n",
      "23\t18154\t0.0\t0\t18154\n",
      "24\t18746\t0.0\t0\t18746\n",
      "25\t17012\t0.0\t0\t17012\n",
      "26\t16100\t0.0\t0\t16100\n",
      "27\t16\t0.0\t0\t16\n",
      "28\t1\t0.0\t0\t1\n",
      "66\t1\t0.0\t0\t1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lauro/anaconda3/envs/qiime2-2023.5/lib/python3.8/site-packages/q2_demux/_summarize/_visualizer.py:191: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  context['result_data'] = context['result_data'].append(df)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lauro/anaconda3/envs/qiime2-2023.5/lib/python3.8/site-packages/q2_demux/_summarize/_visualizer.py:191: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  context['result_data'] = context['result_data'].append(df)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lauro/anaconda3/envs/qiime2-2023.5/lib/python3.8/site-packages/q2_demux/_summarize/_visualizer.py:191: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  context['result_data'] = context['result_data'].append(df)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lauro/anaconda3/envs/qiime2-2023.5/lib/python3.8/site-packages/q2_demux/_summarize/_visualizer.py:191: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  context['result_data'] = context['result_data'].append(df)\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = trim_paired(\n",
    "        demultiplexed_sequences=artifact,\n",
    "        front_f=forward_primer,\n",
    "        front_r=reverse_primer,\n",
    "#         adapter_f=reverse_reverse_complement,\n",
    "#         adapter_r=forward_reverse_complement,\n",
    "        cores=threads,\n",
    "        overlap=trim['overlap'],\n",
    "        indels=False,\n",
    "        match_read_wildcards=True,\n",
    "        match_adapter_wildcards=True,\n",
    "        error_rate=0.01,\n",
    "        discard_untrimmed=True,\n",
    "    ).trimmed_sequences\n",
    "\n",
    "    demux.visualizers.summarize(res).visualization\n",
    "    res.save(demux_file_trim)\n",
    "    Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 570.114377,
   "end_time": "2023-07-04T21:45:37.102297",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/mnt/nupeb/rede-micro/redemicro-miliane-nutri/experiments/miliane-CxAC-trim/nb-executed-steps/step-prepare-data-miliane-CxAC-trim.ipynb",
   "parameters": {
    "base_dir": "/mnt/nupeb/rede-micro/redemicro-miliane-nutri",
    "class_col": "group-id",
    "classifier_file": "/mnt/nupeb/rede-micro/qiime2-classifiers/silva-138-99-nb-classifier.qza",
    "experiment_name": "miliane-CxAC-trim",
    "manifest_file": "/mnt/nupeb/rede-micro/redemicro-miliane-nutri/data/raw/manifest/miliane-manifest-CxAC.csv",
    "metadata_file": "/mnt/nupeb/rede-micro/redemicro-miliane-nutri/data/raw/metadata/miliane-metadata-CxAC.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2023-07-04T21:36:06.987920",
   "version": "2.4.0"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}